src/spikeinterface/core/tests/test_sortinganalyzer.py

```diff
@@ -255,6 +259,28 @@ def test_extension():
         register_result_extension(DummyAnalyzerExtension2)
 
 
+def test_extensions_sorting():
+
+    # nothing happens if all parents are on the left of the children
+    extensions_in_order = {"random_spikes": {"rs": 1}, "waveforms": {"wv": 2}}
+    sorted_extensions_1 = _sort_extensions_by_dependency(extensions_in_order)
+    assert sorted_extensions_1 == extensions_in_order
+
+    extensions_out_of_order = {"waveforms": {"wv": 2}, "random_spikes": {"rs": 1}}
+    sorted_extensions_2 = _sort_extensions_by_dependency(extensions_out_of_order)
+    assert sorted_extensions_2 == extensions_in_order
+
+    # doing two movements
+    extensions_qm_left = {"quality_metrics": {}, "waveforms": {}, "templates": {}}
+    sorted_extensions_3 = _sort_extensions_by_dependency(extensions_qm_left)
+    assert sorted_extensions_3 == {"templates": {}, "waveforms": {}, "quality_metrics": {}}
+
+    # should move parent (waveforms) left of child (quality_metrics), and move grandparent (random_spikes) left of parent
+    extensions_qm_left = {"quality_metrics": {}, "waveforms": {}, "templates": {}, "random_spikes": {}}
+    sorted_extensions_4 = _sort_extensions_by_dependency(extensions_qm_left)
+    assert sorted_extensions_4 == {"templates": {}, "random_spikes": {}, "waveforms": {}, "quality_metrics": {}}
+
```

Comment:
We should have "random_spikes" at the first place no ?

Reply:
Hello. Yes, the tests were all useless since dict equality doesn't care about order. So, at the testing stage, I've made them OrderedDicts.

Reply:
dict is now ordered by default.

Reply:
Yes, but equality doesn't care about the ordering, for backward compatibility. The following

```
a = {1: 3, 2: 4}
b = {2: 4, 1: 3}
a == b
```

outputs `True`

Reply:
yes of course. We should do
list(a.key()) == list(b.keys())
no ?

Reply:
Sure, both seem to work, but if that's standard practice then let's go for keys. I've updated the tests, including getting rid of the "magic" sorted dictionaries.